client/alloc_runner.go

```diff
@@ -436,6 +438,11 @@ func (r *AllocRunner) Run() {
 			return
 		}
 		r.ctx = driver.NewExecContext(allocDir, r.alloc.ID)
+		if r.otherAllocDir != nil {
+			if err := allocDir.Move(r.otherAllocDir, tg.Tasks); err != nil {
+				r.logger.Printf("[ERROR] client: failed to move alloc dir into alloc %q: %v", r.alloc.ID, err)
```

Comment:
@dadgar I wasn't sure if a task event made sense, since an allocation directory isn't task specific.

Regarding building a fresh one, I think we should just log that it has failed and start the allocation dir since it's best effort. Or, we could just make the allocation fail, and server/client could restart it.

I will however ensure that the allocation directory is destroyed.